[DNM] Test of sanitycheck filtering later

galak opened the pull request.

nashif committed:
sanitycheck: Complete overhaul and job handling rework

A complete overhaul of the sanitycheck script and how we build and run
tests. This new version of sanitycheck uses python for job distribution
and drop use of Make.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
scripts: remove obsolete conversion script ini2yaml.py

script is old and unused.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: retry failed in same session

When things fail, retry failed tests in same session as many times as
specified. This produces results for the complete testsuite.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: enable inline logs

Fix --inline-logs option.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: report failure reason

When something fails, report the reason on the console.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: deal with not-installed simulators

When a simulator is not installed, do not attempt to run tests on that
simulator.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
scripts: remove unused waitpid script

Was used previously by sanitycheck, not being used anymore.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: reporting and retries

Improve reporting and support retires.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
vscode: Add laucnh.json for sanitycheck [DO NOT MERGE]

add launchers for vscode to allow debugging.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: support multiple devices

Support testing on multiple connected devices in parallel. New options
added:

--generate-hardware-map map.yml

will scan for connected devices and generate map.yml

then you do

sanitycheck --hardware-map map.yml --device-testing

and will be able to run on multiple devices at the same time.

note: not all types of boards are supported yet, pyocd and nrfjprog
flashable devices work fine.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

galak committed:
sanitycheck: handle ThreadPoolExecutor for python3.5

The ThreadPoolExecutor api in 3.5 only takes a max_workers arg

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: Use same csv format for report and save/load test

Use the same csv format regardless of purpose.  Makes it easier to move
around between states this way.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: add initial test_only support

Add the ability to just do a '--test-only' if we already have a build
setup.  We need access to a <TEST>.csv file either via --load-tests or
we'll look for sanitycheck.csv.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: Handle setting of QEMU_FIFO at runtime

To support --test-only for QEMU target we need to set the qemu_fifo at
run time and not build time.  Since we don't know exactly how the build
we're using was setup, its safer to just pass it at runtime.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif committed:
ci: adapt ci script for new sanitycheck

We now run sanitycheck with different options, so adapt script.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: pylint fixes

Misc. syntax fixes found by pylint.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

galak committed:
sanitycheck: Fix how often we call create_overlay for a TestInstance

We only need to call create_overlay once per TestInstance

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: HACK to filter later

hack some code to handle filtering later.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: Set yamlfile as part of TestCase

The filtering code can reports an error message and uses
self.testcase.yamlfile.  Set testcase.yamlfile when we init a TestCase
so we have access to the filename later to report the error properly.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif commented:
This one is dropping tests:

with this PR:
```
sanitycheck -T tests/kernel/fatal/
Cleaning output directory /home/anashif/zephyrproject/zephyr/sanity-out
JOBS: 112
Selecting default platforms per test case
Building testcase defconfigs...
Filtering test cases...
54 tests selected, 916 tests discarded due to filters
total complete:   16/  54  29%  failed:    0
54 of 54 tests passed with 0 warnings in 10 seconds
In total 5 test cases were executed on 16 out of total 195 platforms (8.21%)
```


with sanitycheck in master:

```
sanitycheck -T tests/kernel/fatal/
JOBS: 112
Cleaning output directory /home/anashif/zephyrproject/zephyr/sanity-out
Selecting default platforms per test case
Building testcase defconfigs...
Filtering test cases...
24 tests selected, 961 tests discarded due to filters
total complete:   24/  24  100%  failed:    0
24 of 24 tests passed with 0 warnings in 14 seconds
```

nashif committed:
ci: run sanitycheck with timestamps

Print timestamps to show where we spend time.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

galak committed:
sanitycheck: Try and make subsets more fair

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: Handle cmake build failures

Handle the case if the cmake fails for some reason and report it as a
build error.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: Remove --config option

We haven't done anything with the --config option for some time and no
one has noticed.  Just remove it as dead code.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif committed:
sanitycheck: more pylint fixes

Apply more pylint fixes.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
this one is already integrated in sanitycheck
